src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,320 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
+                        if (locationCorrect == "")
                         {
-                            IfDiagnostic(context, statements[5], EndSpanIncorrectRule, openParenToken.Text);
+                            IfDiagnostic(context, statements[statementCount - 2], LocationIncorrectRule, statementIdentifierToken.Text);
                             return false;
                         }
 
-                        if (statementCount > 6)
+                        if (statementCount > 5)
                         {
-                            SyntaxToken spanToken = SpanAnalysis(startToken, endToken, statements);
-                            if (spanToken.Text == "")
+                            List<string> spanNames = SpanAnalysis(statements[statementCount - 3], locationCorrect);
+                            if (spanNames.Count == 0 || spanNames[0] == "" || spanNames[1] == "")
                             {
-                                IfDiagnostic(context, statements[6], SpanIncorrectRule, startToken.Text, endToken.Text);
+                                IfDiagnostic(context, statements[statementCount - 3], SpanIncorrectRule, locationCorrect);
                                 return false;
                             }
 
-                            if (statementCount > 7)
+                            if (statementCount > 6)
                             {
-                                SyntaxToken locationToken = LocationAnalysis(statementIdentifierToken, spanToken, statements);
-                                if (locationToken.Text == "")
+                                bool startSpanCorrect = StartAnalysis(keywordIdentifierToken, statements, spanNames[0]);
+                                bool endSpanCorrect = EndAnalysis(statementIdentifierToken, statements, spanNames[1]);
+                                if (!endSpanCorrect && startSpanCorrect)
                                 {
-                                    IfDiagnostic(context, statements[7], LocationIncorrectRule, statementIdentifierToken.Text, spanToken.Text);
+                                    if (statementCount == 7)
+                                    {
+                                        var argument = ((statements[4] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
+                                        ReportDiagnostic(context, EndSpanMissingRule, argument.GetLocation(), locationCorrect, statementIdentifierToken.Text);
+                                        return false;
+                                    }
+                                    else
+                                    {
+                                        var argument = ((statements[5] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
```

Comment:
@jepetty I think that the statement that you grab here should be counted from the end. If there are 15 statements, and a bunch are incorrect, you should grab hte bottom of them not the top. I think.